src/ArborX_LinearBVH.hpp

```diff
@@ -112,9 +112,51 @@ class BasicBoundingVolumeHierarchy
 };
 
 template <typename MemorySpace>
-using BoundingVolumeHierarchy =
-    BasicBoundingVolumeHierarchy<MemorySpace, Details::PairIndexVolume<Box>,
-                                 Details::DefaultIndexableGetter, Box>;
+class BoundingVolumeHierarchy
+    : public BasicBoundingVolumeHierarchy<MemorySpace,
+                                          Details::PairIndexVolume<Box>,
+                                          Details::DefaultIndexableGetter, Box>
+{
+  using base_type =
+      BasicBoundingVolumeHierarchy<MemorySpace, Details::PairIndexVolume<Box>,
+                                   Details::DefaultIndexableGetter, Box>;
+
+public:
+  using legacy_tree = void;
```

Comment:
I'm really not a fan of this, but not sure what else to do.

Essentially, when in the bowels of `CrsGraphWrapperImpl`, we need to know when to wrap the `InsertGenerator` calls. If it's called with `BasicBoundingVolumeHierarchy`, we need to wrap. If with `BoundingVolumeHierarchy`, we don't.

Reply:
We consider it a private tag though right?
Why is it not needed with BruteForce?

Reply:
BruteForce still uses old type of callbacks, and will automatically go to the right operator inside `InsertGenerator`.